### ISSUE TITLE
Error using csv/serde

### ISSUE THREAD

**Issue by lnicola** [paupino/rust-decimal#33]
```
CSV deserialize error: record 1 (line: 2, byte: 523): invalid type: floating point `17.19287`, expected a Decimal type representing a fixed-point number
```

It doesn't seem to be related to parsing, since `Decimal::from_str("17.19287")` works fine.

**Comment by lnicola** [comment]
Repro here: https://github.com/lnicola/decimal-test. Run with `cargo run < foo.csv`.

**Comment by lnicola** [comment]
I don't see anything obviously wrong with the implementation, but I used `serde` only so much. One thing that caught my eye is that the visitor seems to return `0` on error, is that a good idea?

**Comment by paupino** [comment]
Thanks for the feedback. I suspect this is because I've only implemented `visit_i16`, `visit_i32` and `visit_str` methods. Since the CSV has a number without quotes I suspect it thinks it's a float. I'll take a look at implementing this also.

**Comment by lnicola** [comment]
But this works: https://github.com/alkis/decimal/blob/master/src/dec128.rs#L88-L124

**Comment by paupino** [comment]
Hmm, you're quite right. I'll write a few more tests to see what's going on. Thanks for following up!

**Comment by paupino** [comment]
This has now been fixed by #37 and released as part of `0.4.0`. Your tests seem to be passing now.
